src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/SurveySvcClient.java

```diff
@@ -28,6 +100,126 @@
+    SurveyDTO result = null;
+    if (responseEntity != null && responseEntity.getStatusCode().is2xxSuccessful()) {
+      String responseBody = responseEntity.getBody();
+      try {
+        result = objectMapper.readValue(responseBody, SurveyDTO.class);
+      } catch (IOException e) {
+        String msg = String.format("cause = %s - message = %s", e.getCause(), e.getMessage());
+        log.error(msg);
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Get classifier type selector for Survey UUID and ClassifierType UUID.
+   *
+   * @param surveyId UUID for which to request classifiers.
+   * @return SurveyDTO representation of survey
+   * @throws RestClientException something went wrong making http call.
+   */
+  @Retryable(
+      value = {RestClientException.class},
+      maxAttemptsExpression = "#{${retries.maxAttempts}}",
+      backoff = @Backoff(delayExpression = "#{${retries.backoff}}"))
+  public SurveyDTO findSurvey(final UUID surveyId) {
+    UriComponents uriComponents =
+        restUtility.createUriComponents(
+            appConfig.getSurveySvc().getSurveyDetailPath(), null, surveyId);
+
+    HttpEntity<?> httpEntity = restUtility.createHttpEntity(null);
+    ResponseEntity<String> responseEntity;
+    SurveyDTO survey;
+    try {
+      log.with("survey_id", surveyId.toString()).debug("Retrieving survey");
+      responseEntity =
+          restTemplate.exchange(uriComponents.toUri(), HttpMethod.GET, httpEntity, String.class);
+      survey = getSurveyDtoFromResponseEntity(responseEntity);
+    } catch (HttpClientErrorException e) {
+      if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
+        return null;
+      }
+      log.error("Failed to retrieve survey", e);
+      throw e;
+    }
+    return survey;
+  }
+
+  @Retryable(
+      value = {RestClientException.class},
+      maxAttemptsExpression = "#{${retries.maxAttempts}}",
+      backoff = @Backoff(delayExpression = "#{${retries.backoff}}"))
+  public SurveyDTO findSurveyByRef(final String surveyRef) {
+    UriComponents uriComponents =
+        restUtility.createUriComponents(
+            appConfig.getSurveySvc().getSurveyRefPath(), null, surveyRef);
+
+    HttpEntity<?> httpEntity = restUtility.createHttpEntity(null);
+    ResponseEntity<String> responseEntity = null;
+    SurveyDTO survey = null;
+
+    try {
+      responseEntity =
+          restTemplate.exchange(uriComponents.toUri(), HttpMethod.GET, httpEntity, String.class);
+
+      survey = getSurveyDtoFromResponseEntity(responseEntity);
+    } catch (HttpClientErrorException e) {
+      if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
+        return null;
+      }
+      log.error("Failed to retrieve survey", e);
+      throw e;
+    }
+    return survey;
+  }
+
+  public SurveyDTO getSurveyForCollectionExercise(final CollectionExercise collex)
+      throws CTPException {
+    final SurveyDTO survey = findSurvey(collex.getSurveyId());
+
+    if (survey == null) {
+      throw new CTPException(
```

Comment:
does this need to be a checked exception? Personally I'd be in favour of ripping CTPException out of *everywhere*